src/lib/journey/callbacks/text-output/text-output.svelte

```diff
@@ -15,23 +17,31 @@
   // Unused props. Setting to const prevents errors in console
   export const callbackMetadata: Maybe<CallbackMetadata> = null;
   export const selfSubmitFunction: Maybe<SelfSubmitFunction> = null;
-  export const stepMetadata: Maybe<StepMetadata> = null;
+  export let stepMetadata: Maybe<StepMetadata> = null;
   export const style: Style = {};
 
   export let callback: never;
-
-
+  
   let typedCallback = callback as SuspendedTextOutputCallback | TextOutputCallback;
   let dirtyMessage = typedCallback.getMessage();
   let cleanMessage = sanitize(dirtyMessage);
+  let shouldRenderTextOutputNext = false; 
 
   $: {
     typedCallback = callback as SuspendedTextOutputCallback | TextOutputCallback;
     dirtyMessage = typedCallback.getMessage();
     cleanMessage = sanitize(dirtyMessage);
+    shouldRenderTextOutputNext = stepMetadata !== null && (stepMetadata as StepMetadata).isStepSelfSubmittable && !(stepMetadata as StepMetadata).isUserInputOptional
   }
 </script>
 
 <Text classes="tw_font-bold tw_mt-6">
   {@html cleanMessage}
 </Text>
+
+{#if shouldRenderTextOutputNext}
```

Comment:
I'm really hesitant to have a callback component render the "Next" button. I say that because the complex compositional nature of callbacks. One may use the `TextOutputCallback` along with other callbacks. This may result in two, form-submitting buttons being rendered, like this paired with a `ConfirmationCallback`. 

What's more, I wouldn't want this callback to bear the responsibility of it needing awareness of other callbacks, so I think we should have only one place where the "Next" button is rendered and that's at the stage level. 

When I suggested that the stage component shouldn't have to explicitly need to know about callbacks, I didn't necessarily mean that the responsibility should be delegated to the callback component. My suggestion is we might need to improve the generation of the metadata itself to include this use case as a state with which the "Next" button is conditionally rendered in the stage component.

If this doesn't make sense. We can chat about it on Slack or video call. Let me know.